osu.Game/Graphics/UserInterfaceV2/RoundedButton.cs

```diff
@@ -26,19 +36,67 @@ public override float Height
             }
         }
 
+        public override Color4 BackgroundColour
+        {
+            get => base.BackgroundColour;
+            set
+            {
+                base.BackgroundColour = value;
+                triangleGradientSecondColour = BackgroundColour.Lighten(0.2f);
+                updateColours();
+            }
+        }
+
         [BackgroundDependencyLoader(true)]
-        private void load(OsuColour colours)
+        private void load(OverlayColourProvider? overlayColourProvider, OsuColour colours)
         {
             // According to flyte, buttons are supposed to have explicit colours for now.
             // Not sure this is the correct direction, but we haven't decided on an `OverlayColourProvider` stand-in yet.
             // This is a better default. See `SettingsButton` for an override which uses `OverlayColourProvider`.
-            DefaultBackgroundColour = colours.Blue3;
+            DefaultBackgroundColour = overlayColourProvider?.Colour3 ?? colours.Blue3;
+            triangleGradientSecondColour ??= overlayColourProvider?.Colour1 ?? colours.Blue3.Lighten(0.2f);
```

Comment:
Aside from the comment above now being outdated, using `OverlayColourProvider` at this level affects some other screens, such as the editor:

![osu_2022-11-24_22-15-39](https://user-images.githubusercontent.com/20418176/203864003-4fbf6f73-65e9-4c8f-96a8-d28037237f57.jpg)

The buttons here used to be blue, now they're aquamarine-themed. This doesn't look too bad yet, although it's pushing it a bit wrt contrast with text - however it gets worse when you hover:

![osu_2022-11-24_22-15-42](https://user-images.githubusercontent.com/20418176/203864104-45a94316-3a71-40f6-8ec2-7e2d375f1f9e.jpg)

Reply:
This is an edge case that @arflyte is going to have to consider. I think changing the aquamarine colour scheme may be required, as the colours are just brighter than the others.

Again, not hugely fussed about over-brightness when hovered. I don't think it hurts usability too much.

Reply:
I'm leaving this one for now.

Reply:
I'll clean up the outdated comment at nearest convenience in that case.

Reply:
Missed that! Have reworded it.